dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -15,266 +15,181 @@
     }
 
     /**
-     * Check if admin or correct user
-     *
-     * @param user the user that is requesting something
-     * @param id
+     * Check if a user owns (is one of the users that controls) the specified entry.
+     * @param user user to be checked, null if user not logged in
+     * @param entry entry to be checked
      */
-    default void checkUser(User user, long id) {
-        if (!user.getIsAdmin() && user.getId() != id) {
-            throw new CustomWebApplicationException("Forbidden: please check your credentials.", HttpStatus.SC_FORBIDDEN);
-        }
+    default void checkIsOwner(User user, Entry<?, ?> entry) {
+        throwIf(!isOwner(user, entry), FORBIDDEN_ENTRY_MESSAGE, HttpStatus.SC_FORBIDDEN);
     }
 
     /**
-     * Check if user is null
-     *
-     * @param user user to check if null
+     * Check if a user has adminitrative privileges.
+     * @param user user to be checked, null if user not logged in
      */
-    default void checkUserExists(User user) {
-        if (user == null) {
-            throw new CustomWebApplicationException("User not found.", HttpStatus.SC_NOT_FOUND);
-        }
+    default void checkIsAdmin(User user) {
+        throwIf(!isAdmin(user), FORBIDDEN_ADMIN_MESSAGE, HttpStatus.SC_FORBIDDEN);
     }
 
     /**
-     * Check if entry belongs to user
-     *
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for
+     * Check if a the ID of a user matches the specified ID.
+     * @param user user to be checked, null if user not logged in
+     * @param userId id to match
      */
-    default void checkUserOwnsEntry(User user, Entry entry) {
-        if (entry.getUsers().stream().noneMatch(u -> ((User)(u)).getId() == user.getId())) {
-            throw new CustomWebApplicationException("Forbidden: you do not have the credentials required to access this entry.",
-                    HttpStatus.SC_FORBIDDEN);
-        }
-    }
-
-    static boolean userCannotRead(User user, Entry entry) {
-        return !user.getIsAdmin() && (entry.getUsers()).stream().noneMatch(u -> ((User)(u)).getId() == user.getId());
+    default void checkUserId(User user, long userId) {
+        throwIf(user == null || user.getId() != userId, FORBIDDEN_ID_MISMATCH_MESSAGE, HttpStatus.SC_FORBIDDEN);
     }
 
     /**
-     * Checks if a user can read an entry. Default implementation
-     * is to invoke <code>checkUser</code>. Implmenations that support
-     * more nuanched sharing should override.
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
+     * Check if a specified entry exists (is not null).
+     * @param entry entry to be checked
      */
-    default void checkUserCanRead(User user, Entry entry) {
-        checkUser(user, entry);
+    default void checkExistsEntry(Entry<?, ?> entry) {
+        throwIf(entry == null, "Entry not found.", HttpStatus.SC_NOT_FOUND);
```

Comment:
I'm personally not a fan of us using the word "entry"  publicly, even though we already do it in at least one case. See #4908 and the discussion it links to. But the links just show the disagreement, and looking where you invoke this from, it may not be trivial to easily know the type of entry, so I guess it works.

Reply:
I'm pro-saying-entry in most things, since the way I see it a Dockstore entry is more than just the workflow/tool file itself and "entry" vs "workflow/tool" helps emphasize that... but if this error displays publicly to the user, "workflow/tool not found" might be more helpful until/unless we start using the word "entry" more often.

Reply:
Although I agree, this is a relatively difficult fix, since we're handling the `null` case, and thus cannot infer the type from the passed object.  If the UI is operating correctly, the user should rarely see this message, and the behavior is the same as it's been for a long time.  So, I recommend we leave it as is, and focus our efforts elsewhere.